### PR TITLE
Migrate to edition 2018

### LINKED ISSUE #360: Migrate to edition=2018
- [x] Libraries
- [x] Examples

### PR DESCRIPTION
**Pull request by ascjones**
Closes #360.

On each crate:
- ran `cargo fix --edition --allow-dirty`
- added `edition = "2018"` to Cargo.toml
- ran `cargo fix --edition-idioms --broken-code`
- fixed any errors/warning and removed whitespaces

### CODE REVIEW

File: core/src/calls.rs [hunk 1]
```diff
@@ -41,15 +41,15 @@ pub trait RpcNotification<T: Metadata>: Send + Sync + 'static {
 #[derive(Clone)]
 pub enum RemoteProcedure<T: Metadata> {
 	/// A method call
-	Method(Arc<RpcMethod<T>>),
+	Method(Arc<dyn RpcMethod<T>>),
```
Comment (tomusdrw) [inline core/src/calls.rs]: Is this really required now?

File: core/src/calls.rs [hunk 2]
```diff
@@ -41,15 +41,15 @@ pub trait RpcNotification<T: Metadata>: Send + Sync + 'static {
...
 	/// A notification
-	Notification(Arc<RpcNotification<T>>),
+	Notification(Arc<dyn RpcNotification<T>>),
 	/// An alias to other method,
 	Alias(String),
 }
 
 impl<T: Metadata> fmt::Debug for RemoteProcedure<T> {
-	fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+	fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
```
Comment (tomusdrw) [inline core/src/calls.rs]: Seems redundant as well

File: http/src/lib.rs [hunk 3]
```diff
@@ -540,7 +540,7 @@ fn configure_port(reuse: bool, tcp: &net2::TcpBuilder) -> io::Result<()> {
 	use net2::unix::*;
 
 	if reuse {
-		try!(tcp.reuse_port(true));
+		r#try!(tcp.reuse_port(true));
```
Comment (tomusdrw) [inline http/src/lib.rs]: let's convert to `?`

File: core/src/calls.rs (hunk 1 above)
Comment (ascjones) [reply core/src/calls.rs]: Don't think so, just `cargo fix` being overzealous

**Pull request merged**